pkg/apis/componentconfig/types.go

```diff
@@ -355,3 +355,161 @@ type LeaderElectionConfiguration struct {
+	DeploymentControllerSyncPeriod unversioned.Duration `json:"deploymentControllerSyncPeriod"`
+	// podEvictionTimeout is the grace period for deleting pods on failed nodes.
+	PodEvictionTimeout unversioned.Duration `json:"podEvictionTimeout"`
+	// deletingPodsQps is the number of nodes per second on which pods are deleted in
+	// case of node failure.
+	DeletingPodsQps float32 `json:"deletingPodsQps"`
+	// deletingPodsBurst is the number of nodes on which pods are bursty deleted in
+	// case of node failure. For more details look into RateLimiter.
+	DeletingPodsBurst int `json:"deletingPodsBurst"`
+	// nodeMontiorGracePeriod is the amount of time which we allow a running node to be
+	// unresponsive before marking it unhealty. Must be N times more than kubelet's
+	// nodeStatusUpdateFrequency, where N means number of retries allowed for kubelet
+	// to post node status.
+	NodeMonitorGracePeriod unversioned.Duration `json:"nodeMonitorGracePeriod"`
+	// registerRetryCount is the number of retries for initial node registration.
+	// Retry interval equals node-sync-period.
+	RegisterRetryCount int `json:"registerRetryCount"`
+	// nodeStartupGracePeriod is the amount of time which we allow starting a node to
+	// be unresponsive before marking it unhealty.
+	NodeStartupGracePeriod unversioned.Duration `json:"nodeStartupGracePeriod"`
+	// nodeMonitorPeriod is the period for syncing NodeStatus in NodeController.
+	NodeMonitorPeriod unversioned.Duration `json:"nodeMonitorPeriod"`
+	// serviceAccountKeyFile is the filename containing a PEM-encoded private RSA key
+	// used to sign service account tokens.
+	ServiceAccountKeyFile string `json:"serviceAccountKeyFile"`
+	// enableProfiling enables profiling via web interface host:port/debug/pprof/
+	EnableProfiling bool `json:"enableProfiling"`
+	// clusterName is the instance prefix for the cluster.
+	ClusterName string `json:"clusterName"`
+	// clusterCIDR is CIDR Range for Pods in cluster.
+	ClusterCIDR string `json:"clusterCIDR"`
+	// allocateNodeCIDRs enables CIDRs for Pods to be allocated and set on the
+	// cloud provider.
+	AllocateNodeCIDRs bool `json:"allocateNodeCIDRs"`
+	// rootCAFile is the root certificate authority will be included in service
+	// account's token secret. This must be a valid PEM-encoded CA bundle.
+	RootCAFile string `json:"rootCAFile"`
+	// kubeAPIQPS is the QPS to use while talking with kubernetes apiserver.
+	KubeAPIQPS float32 `json:"kubeAPIQPS"`
+	// kubeAPIBurst is the burst to use while talking with kubernetes apiserver.
+	KubeAPIBurst int `json:"kubeAPIBurst"`
+	// leaderElection defines the configuration of leader election client.
+	LeaderElection LeaderElectionConfiguration `json:"leaderElection"`
+	// vloumeConfiguration holds configuration for volume related features.
+	VolumeConfiguration VolumeConfiguration `json:"volumeConfiguration"`
+}
+
+// VolumeConfiguration contains *all* enumerated flags meant to configure all volume
```

Comment:
Struct doc is not included in generated documentation, only field doc is. Not sure whether to leave this capitalized or not. I'm leaving it capitalized because it does not refer to a json field name.